openmc/capi/math.py

```diff
@@ -182,7 +217,57 @@ def calc_zn_rad(n, rho):
     zn_rad = np.zeros(num_bins, dtype=np.float64)
     _dll.calc_zn_rad_c(n, rho, zn_rad)
     return zn_rad
-    
+
+
+def evaluate_zernike_rad(data, r):
+    """ Finds the value of f(x) given a set of even order Zernike coefficients
+    and the value of r.
+
+    Parameters
+    ----------
+    data : iterable of float
+        Legendre coefficients
+    r : float
+        Independent variable to evaluate the radial part of Zernike at
+
+    Returns
+    -------
+    float
+        Corresponding Zernike expansion result
+
+    """
+
+    data_arr = np.array(data, dtype=np.float64)
+    n = int(2 * (len(data) - 1))
+    return _dll.evaluate_zernike_rad_c(n,
+                                        data_arr.ctypes.data_as(POINTER(
+                                        c_double)),
+                                        r)
```

Comment:
Again, I don't think we need an extra C++ function for this. You could use the existing `calc_zn_rad` to get the evaluated polynomials and then perform the element-wise multiplication and sum using standard numpy functionality.

Thinking a little further along the same lines as before, it might be good to create a class in my proposed `openmc/polynomial.py` for Zernike expansion that would look something like:
```Python
class ZernikeRadial:
    def __init__(self, coefficients):
        ...

    def __call__(self, r):
        # Evaluate Zernike polynomials at radius
        # Multiply by coefficients and normalization
        # return sum
```